# Command limits
cmd_max_velocity: 50        # max velocity [0-256], max duty cycle of the PWM controlling the DC motor
cmd_max_angle: 70           # max servo angle [deg]
cmd_max_turn: 20            # max angle step for the servo at each execution [deg] 
cmd_max_acc_step: 1         # max acceleration step at each execution
cmd_max_dec_step: 4         # max deceleration step at each execution
cmd_start_acc_step: 20      # initial velocity step
cmd_zero_speed_tol: 3       # maximum velocity approximated to 0
inertia_stop_step: 15       # simulated step of the stop by inertia (driver in High impedance mode)

# Move2Goal parameters
tolerance: 0.2   # [m]  
k_rho: 1.0       # velocity gain
k_alpha: 2.2     # steering angle gain
k_beta: 0.9       # smooth rotations

# Velocity controller
kp_vel: 1.5             # proportional gain
ki_vel: 0.4             # integral action

# Psi controller
kp_psi: 3.0             # proportional gain
ki_psi: 0.3             # integral action

# Physical parameters of the car-like robot
max_speed: 1.5      # [m/s] referred to the maximum velocity command
max_psi: 19.9        # [deg]
curvature: 1.4      # [m]
wheelbase: 0.38     # [m]

#DC motor parameters
u_r_motor: 0.915              # dynamic friction relatic coefficient
T_load_static: 0.17           # [N*m], modeled as static friction
beta_viscous: 0.00012         # [N*s/rad], viscous friction coefficient
k_speed_motor: 0.0205         # [m/rad], gain between w and v
ta_motor : 0.005              # [s], armature time constant of the DC motor
tm_motor : 3.2                # [s], mechanical time constant of the DC motor
kt_motor : 0.056              # [N*m/V], Kt/Ra gain between torque and voltage
ka_motor: 0.07                # [V], gain between command and voltage  ~18 V/256
kv_motor: 0.0095              # [V*s/rad], back EMF gain

# System configuration
repeat_rate: 10               # [hz]
show_animation: True